### PR TITLE
[Runtime] Enable the -Wsign-conversion warnings flag and fix some of the warnings.

### PR DESCRIPTION
**Pull request by mandel-macaque**
The flag adds a lot (A LOT) of warnings. The diff would be huge and
dangerous to review. We enable the flag and fix file by file.

### CODE REVIEW

File: runtime/trampolines-varargs.m [hunk 1]
```diff
@@ -18,6 +22,7 @@
 #include "runtime-internal.h"
 #include "trampolines-varargs.h"
 
+
```
Comment (whitneyschmidt) [inline runtime/trampolines-varargs.m]: extra line?

Comment (mandel-macaque) [reply runtime/trampolines-varargs.m]: yes, I can remove it :)

File: runtime/mono-runtime.h.t4 [hunk 2]
```diff
@@ -30,7 +30,7 @@ typedef guint    (*GHashFunc)      (gconstpointer key);
 typedef gboolean (*GEqualFunc)     (gconstpointer a, gconstpointer b);
 
 #ifndef GPOINTER_TO_INT
-#define GPOINTER_TO_INT(p) ((int)  (long) (p))
+#define GPOINTER_TO_INT(p) ((uint32_t)  (long) (p))
```
[the remark this answers is not in the document]
Comment (spouliot) [reply runtime/mono-runtime.h.t4]: make sense, but the above `#ifndef` needs to be updated too
i.e. don't commit suggestion alone :)

[the remark this answers is not in the document]
Comment (mandel-macaque) [reply runtime/mono-runtime.h.t4]: I can't it would not compile. It is a larger change (is used in several places) but can be done in this PR.

**Comment by mandel-macaque** [comment]
HE0038 (https://github.com/xamarin/maccore/issues/581) and issues with the msdn page that are fixed by https://github.com/xamarin/xamarin-macios/pull/7401

**Pull request merged**